airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_excel_parser.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import datetime
+from io import BytesIO
+from unittest.mock import Mock, mock_open, patch
+
+import pandas as pd
+import pytest
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import ExcelFormat, FileBasedStreamConfig, ValidationPolicy
+from airbyte_cdk.sources.file_based.exceptions import ConfigValidationError, RecordParseError
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.file_types.excel_parser import ExcelParser
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+
+
+@pytest.fixture
+def mock_stream_reader():
+    return Mock(spec=AbstractFileBasedStreamReader)
+
+
+@pytest.fixture
+def mock_logger():
+    return Mock()
+
+
+@pytest.fixture
+def file_config():
+    return FileBasedStreamConfig(
+        name="test.xlsx",
+        file_type="excel",
+        format=ExcelFormat(sheet_name="Sheet1"),
+        validation_policy=ValidationPolicy.emit_record
+    )
+
+
+@pytest.fixture
+def remote_file():
+    return RemoteFile(uri="s3://mybucket/test.xlsx", last_modified=datetime.datetime.now())
+
+
+@pytest.fixture
+def sample_excel_data():
+    data = {
+        'A': [1, 2, 3],
+        'B': ['2023-01-01', '2023-02-01', '2023-03-01'],
+        'C': [True, False, True]
+    }
+    df = pd.DataFrame(data)
+    with BytesIO() as buffer:
+        with pd.ExcelWriter(buffer, engine='xlsxwriter') as writer:
+            df.to_excel(writer, index=False)
+        return buffer.getvalue()
+
+
+@pytest.mark.asyncio
+async def test_infer_schema(mock_stream_reader, mock_logger, file_config, remote_file, sample_excel_data):
```

Comment:
When running this using PyCharm test runner, I get `Skipped: async def function and no async plugin installed (see warnings)`. Running locally with `python -m pytest -s unit_tests -k test_infer_schema`, I get the following:
```
  /Users/maxime/devel/code/airbyte/airbyte-cdk/python/.venv/lib/python3.10/site-packages/_pytest/python.py:183: PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.
  You need to install a suitable plugin for your async framework, for example:
    - anyio
    - pytest-asyncio
    - pytest-tornasync
    - pytest-trio
    - pytest-twisted
    warnings.warn(PytestUnhandledCoroutineWarning(msg.format(nodeid)))
```

Using `poetry run poe unit-test-with-cov`, I get `unit_tests/sources/file_based/file_types/test_excel_parser.py::test_infer_schema SKIPPED (async def function and no async plugin installed (see warnings))`.

So all seems to indicate that this test does not run. Is there a dependency missing that you would have locally that enabled you to test that?

Reply:
It's an interesting situation because the `test_csv_parser` imports asyncio at the very beginning, but it's not mentioned in the dependency list. I've added the asyncio package to the Poetry dependencies list.